app/src/sharedTest/java/org/oppia/app/utility/TabMatcher.kt

```diff
@@ -0,0 +1,34 @@
+package org.oppia.app.utility
+
+import android.view.View
+import org.hamcrest.Description
+import org.hamcrest.TypeSafeMatcher
+import androidx.test.espresso.PerformException
+import com.google.android.material.tabs.TabLayout
+import org.hamcrest.Matcher
+import java.lang.IllegalStateException
+
+/**
+ * A custom matcher to test whether the current tab title is correctly selected in a [TabLayout].
+ *
+ * Reference: https://stackoverflow.com/q/49626315 */
+class TabMatcher {
+  companion object {
+    fun matchCurrentTabTitleCheck(tabTitle: String): Matcher<View> {
+      return object : TypeSafeMatcher<View>() {
+        override fun describeTo(description: Description?) {
+          description?.appendText("unable to match title of current selected tab with $tabTitle")
+        }
+
+        override fun matchesSafely(item: View?): Boolean {
+          val tabLayout = item as TabLayout
+          val tabAtIndex: TabLayout.Tab = tabLayout.getTabAt(tabLayout.selectedTabPosition)
+            ?: throw PerformException.Builder()
+              .withCause(throw IllegalStateException("No tab at index ${tabLayout.selectedTabPosition}"))
```

Comment:
This code is giving `Unreachable code` warning can you fix that.

Reply:
throw IllegalStateException
Suggested by ben

Reply:
Resolved